include/pops/kernel_base.hpp

```diff
@@ -0,0 +1,131 @@
+     */
+    virtual std::tuple<int, int> operator()(Generator& generator, int row, int col) = 0;
+
+    virtual bool is_cell_eligible(int row, int col) = 0;
+
+    /*! \copydoc RadialDispersalKernel::supports_kernel()
+     */
+    virtual bool supports_kernel(const DispersalKernelType type) = 0;
+
+    virtual ~KernelInterface() = default;
+};
+
+/**
+ * Base class for common implementation of operator() and other methods.
+ *
+ * Methods optional in static such as is_cell_eligible kernels need to be only in
+ * derived classes because valid implementation is required even if the method is
+ * eventually overridden.
+ */
+template<typename ActualKernel, typename Generator>
+class BaseDynamicKernel : public KernelInterface<Generator>
+{
+public:
+    BaseDynamicKernel(const ActualKernel& kernel) : kernel_(kernel) {}
+
+    template<typename... Args>
+    BaseDynamicKernel(Args&&... args) : kernel_(std::forward<Args>(args)...)
+    {}
+
+    /*! \copydoc RadialDispersalKernel::operator()()
+     */
+    std::tuple<int, int> operator()(Generator& generator, int row, int col) override
+    {
+        return kernel_.operator()(generator, row, col);
+    }
+
+    /*! \copydoc RadialDispersalKernel::supports_kernel()
+     */
+    bool supports_kernel(const DispersalKernelType type) override
+    {
+        return ActualKernel::supports_kernel(type);
+    }
+
+protected:
+    ActualKernel kernel_;
+};
+
+/**
+ * Default class which turns a any kernel into a kernel usable with KernelInterface.
+ *
+ * Kernel needs to implement is_cell_eligible.
+ */
+template<typename ActualKernel, typename Generator>
+class DynamicKernel : public BaseDynamicKernel<ActualKernel, Generator>
+{
+public:
+    DynamicKernel(const ActualKernel& kernel)
+        : BaseDynamicKernel<ActualKernel, Generator>(kernel)
+    {}
+
+    template<typename... Args>
+    DynamicKernel(Args&&... args)
+        : BaseDynamicKernel<ActualKernel, Generator>(std::forward<Args>(args)...)
+    {}
+
+    bool is_cell_eligible(int row, int col) override
+    {
+        return BaseDynamicKernel<ActualKernel, Generator>::kernel_.is_cell_eligible(
+            row, col);
+    }
+};
+
+/**
+ * Class which turns a any kernel into a kernel usable with KernelInterface.
+ *
+ * Useful for kernles which do not implement is_cell_eligible.
+ */
+template<typename ActualKernel, typename Generator>
+class AlwaysEligibleDynamicKernel : public BaseDynamicKernel<ActualKernel, Generator>
+{
+public:
+    AlwaysEligibleDynamicKernel(const ActualKernel& kernel)
+        : BaseDynamicKernel<ActualKernel, Generator>(kernel)
+    {}
+
+    template<typename... Args>
+    AlwaysEligibleDynamicKernel(Args&&... args)
+        : BaseDynamicKernel<ActualKernel, Generator>(std::forward<Args>(args)...)
+    {}
+
+    bool is_cell_eligible(int row, int col) override
+    {
+        UNUSED(row);
+        UNUSED(col);
+        return true;
+    }
+};
+
+}  // namespace pops
```

Comment:
After discussion, let's implement `is_cell_eligible` in all kernels instead of the complex code.

Reply:
Done. Now only one concrete class is needed.